sdk/core/Azure.Core.Experimental/src/RequestOptions.cs

```diff
@@ -52,6 +97,20 @@ public class RequestOptions
         /// </summary>
         public HttpPipelinePolicy? PerCallPolicy { get; set; }
 
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="message"></param>
+        public void Apply(HttpMessage message)
```

Comment:
Why is this public?

Reply:
It implies that is applies all the options, and it does not.

Reply:
We would replace https://github.com/Azure/autorest.csharp/blob/73f84288492133555f6928c5c6875d2cf6141e03/test/TestServerProjectsLowLevel/httpInfrastructure/Generated/HttpClientFailureClient.cs#L98-L101 with `options.Apply();` call

Reply:
Can we move the method somewhere else? into a non-consume facing type.